modules/phase_field/src/kernels/MultiGrainRigidBodyMotion.C

```diff
@@ -0,0 +1,80 @@
+/****************************************************************/
+/* MOOSE - Multiphysics Object Oriented Simulation Environment  */
+/*                                                              */
+/*          All contents are licensed under LGPL V2.1           */
+/*             See LICENSE for full restrictions                */
+/****************************************************************/
+#include "MultiGrainRigidBodyMotion.h"
+
+template<>
+InputParameters validParams<MultiGrainRigidBodyMotion>()
+{
+  InputParameters params = validParams<Kernel>();
+  params.addClassDescription("Adds rigid mody motion to grains");
+  params.addRequiredCoupledVar("c", "Concentration");
+  params.addParam<MaterialPropertyName>("advection_velocity", "Material property for advection velocities");
```

Comment:
You need to make thos params either required, or provide a default, or check if they are valid before you use them.
My preference is providing a default value! And make the defaults consistent across the classes to that the users won't have to specify the parameter at all in most cases.